automation/get_target_servers.py

```diff
@@ -0,0 +1,144 @@
+    """Prints the error returned from an API call"""
+    return f'Error: {response.status_code} - {response.reason}. \n {response.text}'
+
+
+def parse_args():
+    """Defines which arguments are needed for the script to run."""
+    parser = argparse.ArgumentParser(
+        description='Gets target server by name from Apigee')
+    req_grp = parser.add_argument_group(title='required arguments')
+    req_grp.add_argument(
+        '-tsn',
+        '--target_server_name',
+        help='name of the target server',
+        required=True)
+    req_grp.add_argument(
+        '-env',
+        '--environments',
+        help='names of the environments',
+        required=True)
+    req_grp.add_argument(
+        '-org',
+        '--organizations',
+        help='names of the organizations',
+        required=True)
+    req_grp.add_argument(
+        '-u',
+        '--username',
+        help='apigee user')
+    req_grp.add_argument(
+        '-pwd',
+        '--password',
+        help='apigee password')
+    req_grp.add_argument(
+        '-rt',
+        '--refresh_token',
+        help='apigee refresh token')
+    req_grp.add_argument(
+        '-out',
+        '--output',
+        help='file location of the query output',
+        required=False)
+
+    parsed = parser.parse_args()
+
+    if parsed.refresh_token is None and (parsed.username is None or parsed.password is None):
+        parser.error("the following arguments are required: -u/--username and -pwd/--password OR -r/--refresh_token")
+
+    return parsed
+
+
+def main():
+    """Method called from the main entry point of the script to do the required logic."""
+    args = parse_args()
+
+    target_server_name = args.target_server_name
+    environments = args.environments.split(',')
+    organizations = args.organizations.split(',')
+    username = args.username
+    password = args.password
+    refresh_token = args.refresh_token
+    output_path = args.output
+
+    # Retrieve an access token using the refresh token provided. This ensures
+    # that we always have a valid access token.
+    if refresh_token is not None:
+        access_token = apigee_auth.refresh_access_token(refresh_token)
+    else:
+        access_token = apigee_auth.get_access_token(username, password)
+
+    # Add Auth Header by default to all requests.
+    REQUEST.headers.update({'Authorization': 'Bearer {}'.format(access_token)})
+
+    result = []
+
+    for org in organizations:
+        for env in environments:
+            if not exists_env_in_org(org, env):
+                continue
+
+            target_server = get_target_server(org, env, target_server_name)
+            result.append(TargetServer(org, env, target_server))
+
+    for r in result:
+        print(r)
+
+    z = zipfile.ZipFile(io.BytesIO(result))
+    z.extractall(output_path)
+
+    location = output_path + '/' + z.filelist[0].filename
+    print('File stored in: {}'.format(location))
+
+    return location
```

Comment:
The output path is an optional argument. If it isn't present, this entire section is not necessary. This can be wrapped in a conditional statement and only executed when the output path is provided.